src/mapgen.cpp

```diff
@@ -589,13 +589,13 @@ class jmapgen_alternativly : public jmapgen_piece {
     // PieceType, they *can not* be of any other type.
     std::vector<PieceType> alternatives;
     jmapgen_alternativly() = default;
-    void apply( map &m, const size_t x, const size_t y, const float mon_density ) const override
+    void apply( map &m, const size_t x, const size_t y, const size_t xmin, const size_t ymin, const size_t xmax, const size_t ymax, const float mon_density ) const override
```

Comment:
That would probably be better.
Currently all functions use either 2 coords (x/y) or the other 4, not all 6.